BPA/Ch.10/js_super_indran.stan

```diff
@@ -231,24 +250,22 @@ generated quantities {
     int Nind[M];
     int Nalive[M];
 
+    for (i in 1:M)
+      for (t in 1:n_occasions)
+        u[i, t] = z[i, t] * w[i];
     for (i in 1:M) {
-      for (t in 1:n_occasions) {
-        u[i, t] <- z[i, t] * w[i];
-      }
-    }
-    for (i in 1:M) {
-      recruit[i, 1] <- u[i, 1];
+      recruit[i, 1] = u[i, 1];
       for (t in 2:n_occasions)
-        recruit[i, t] <- (1 - u[i, t - 1]) * u[i, t];
-    } //i
+        recruit[i, t] = (1 - u[i, t - 1]) * u[i, t];
```

Comment:
These can also be done with elementwise products and slicing as

```
recruit[i, 1] = u[i, 1];
recruit[i, 2:n_occassions] = (1 - u[i, 1:(n_occassions - 1)]) .* u[i, 2:n_occasions];
```

You'll want to double check I got my algebra/indexing right if you want to make these changes.